.gitmodules

```diff
@@ -1,4 +1,4 @@
 [submodule "crypto"]
 	path = crypto
 	url = https://github.com/ARMmbed/mbed-crypto
-	branch = development
+	branch = iotssl-2697-psa-key-allocation-crypto
```

Comment:
I'm not a git submodule expert, but I was expecting this line to remain unchanged and instead checking out https://github.com/ARMmbed/mbed-crypto/pull/22/commits/54c139f1418937588098b0435dbd8de85fe40223 in the submodule. @mpg ?

Reply:
I agree, we don't need (or want to) change this file. The proper way to point the submodule to a specific commit is:
```
cd crypto
git checkout <commit>
cd ..
git commit crypto
```
The `branch` entry in `.gitmodules` is only used by other git commands, not by `git submodule update` nor `git clone --recursive`.